src/components/app/App.js

```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+import PaymentsDetails from '../payments-details/paymentsDetails';
+import {BrowserRouter, Switch} from 'react-router-dom';
+import MainContent from '../main_content/mainContent'
+
+
+import {Route} from 'react-router-dom';
+
+import './app.css';
+
+
+export default class App extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            users: [],
+            spendings: [],
```

Comment:
while it's not that critical but still don't forget about proper formatting of your code. this one should look like:

```javascript
        this.state = {
            users: [],
            spendings: [],
            sumOfGroupSpent : 0
        };
```